baremetal/metal3machine_manager.go

```diff
@@ -150,6 +153,68 @@ func (m *MachineManager) role() string {
 	return bmRoleNode
 }
 
+// CheckPauseAnnotation checks and/or Removes the pause annotations on associated bmh
+func (m *MachineManager) CheckPauseAnnotation(ctx context.Context) error {
```

Comment:
the naming of the function might be a bit misleading, when its purpose is to remove the annotation conditionally.

Reply:
Its not removing in case it is set by the user, thats why it is check, but I can rename it.